tensorflow/python/kernel_tests/segment_reduction_ops_test.py

```diff
@@ -30,28 +30,39 @@ def _input(self, input_shape, dtype=tf.int32):
       num_elem *= x
     values = np.arange(1, num_elem + 1)
     np_values = values.reshape(input_shape).astype(dtype.as_numpy_dtype)
+
     return tf.constant(values, shape=input_shape,
                                 dtype=dtype), np_values
 
-  def _segmentReduce(self, indices, x, op1, op2=None, num_out_rows=None):
-    if not x.size: return np.array([])
-    indices = np.asarray(indices)
-    if num_out_rows is None:
-      num_out_rows = indices[-1] + 1
-    output = [None] * num_out_rows
-    slice_shape = x.shape[indices.ndim:]
-    x_flat = x.reshape((indices.size,) + slice_shape)
-    for i, index in enumerate(indices.ravel()):
-      if output[index] is not None:
-        output[index] = op1(output[index], x_flat[i])
+  def _segmentReduce(self, indices, x, op1, op2=None, num_out_rows=None,dtype=tf.int32):
+      if not x.size: return np.array([])
+      indices = np.asarray(indices)
+      if num_out_rows is None:
+          num_out_rows = indices[-1] + 1
+      output = [None] * num_out_rows
+      slice_shape = x.shape[indices.ndim:]
+      x_flat = x.reshape((indices.size,) + slice_shape)
+      for i, index in enumerate(indices.ravel()):
+          if (output[index] is not None) and op1 == np.max:
+
+              for j in range(0,output[index].shape[0]):
+                  output[index][j] = op1([output[index][j], x_flat[i][j]])
+          elif output[index] is not None:
+              output[index] = op1(output[index], x_flat[i])
+          else:
+              output[index] = x_flat[i]
+      # zero initialize values that are still uncalcuated.
+      # output = [o if o is not None else np.zeros(slice_shape) for o in output]
+      if not op1 == np.max:
```

Comment:
Nit: perhaps clearer to condition as 
if op1 == np.max
instead of conditioning on not of that.